### PR TITLE
nos.social through Traefik

### PR DESCRIPTION
**Pull request by dcadenas**
Adds a similar setup to relay.nos.social for nos.social.

### CODE REVIEW

File: inventories/nos-social/inventory.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+nos-social:
```
Comment (cooldracula) [inline inventories/nos-social/inventory.yml:2]: This group name may cause weird errors.  Python(and thus Ansible) want snake_case and I find having inventory groups with a dash instead of underscore can cause issues.  I recommend changing our instances of `nos-social` to `nos_social` within this file, along with the name of the inventory, role, and playbook.

File: inventories/nos-social/inventory.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  hosts:
+    nos-social2.ansible.fun:
+  vars:
+    admin_username: admin
+    homedir: /home/{{ admin_username }}
+    cert_email: zach@nos.social
+    domain: '{{ inventory_hostname }}'
```
Comment (cooldracula) [inline inventories/nos-social/inventory.yml:9]: Do we need the relay_image_tag variable?

File: inventories/nos-social/inventory.yml.bkup_23-12-05_09:53 [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (cooldracula) [inline inventories/nos-social/inventory.yml.bkup_23-12-05_09:53:1]: This file is a bit misleading.  It's created by our new-do-droplet  play when it creates a new server.  As part of the creation, we update our inventory file.  Because of how we have to do the update (essentially writing it again with the new values) I wanted to make sure we did a backup before the update, so that we can revert to the backup in case there's an issue and our inventory is not lost.  If everything goes to plan, then this backup file can be deleted.  

In other words, we shouldn't commit this file.  It can be deleted once you verified that the inventory update was as you expected.  We should probably add `inventory*.bkup_*` to our gitignore.  But this file, and the one below, can be deleted.

File: inventories/nos-social/inventory.yml.bkup_23-12-05_10:06 [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (cooldracula) [inline inventories/nos-social/inventory.yml.bkup_23-12-05_10:06:1]: same as above

**Review by cooldracula — CHANGES_REQUESTED** [review]
This generally looks good!  I requested changes, but they mostly have to do with ansible gotchas, and not with the work itself.  This role nicely follows the pattern of your previous PR's and our other roles.

I _believe_ this role is in a transition state at the moment, yah, as there is an api service that will be added that is being developed separately?  I think we should review and merge this when that additional service is added.  At the moment, it seems like the deployed service just redirects all traffic to webflow, and so is sort of redundant.  Am I understanding the role and context correctly?

**Comment by dcadenas** [comment]
@cooldracula, thank you for your input. I've updated the branch based on your suggestions. You were correct in your understanding that the goal is to deploy a service within one of the Traefik containers. Specifically, I plan to host the service found at [nip05api](https://github.com/planetary-social/nip05api), which appears to be ready for deployment.

This service requires a Redis instance. At this point, I'm contemplating whether to run Redis on the same host as a named volume or to dedicate a separate host exclusively for Redis. For now, I'm leaning towards starting with the Redis instance on the same host for simplicity. We can consider migrating it to a dedicated host if we observe significant growth or resource demands.

**Review by cooldracula — APPROVED** [review]
Thank you for these changes, it looks good!

There seemed to be a duplicate example in new-server-vars.yml for nos.social, so I removed it and pushed that change.  

I think this is good to merge!
